#! /bin/sh -e

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

# DP: On sparc default to v9 in 32bit mode

--- gcc/config.gcc~	2009-10-18 03:09:56.000000000 +0200
+++ gcc/config.gcc	2009-10-21 14:44:57.000000000 +0200
@@ -3156,6 +3156,13 @@
 			;;
 		esac
 
+		# setting this directly fails to build a biarch defaulting to 32bit
+		case "$target" in sparc-*-linux*)
+			if test "$option" = cpu && test "$val" = v7; then
+				val=v9
+			fi
+		esac
+
 		if test "x$t" = x
 		then
 			t="{ \"$option\", \"$val\" }"
